game/player.asm

```diff
@@ -0,0 +1,145 @@
+.section "Player" BANK 0 SLOT "ROM"
+
+nop
+
+.struct Player
+    oam_manager_ptr instanceof OAMManager
+    input instanceof Input
+    oam_obj_ptr dw ; Pointer to the requested OAM object
+    ; enabled db
+.endst
+
+.enum $0000
+    player instanceof Player
+.ende
+
+Player_Init:
+    pha
+    phy
+    phx
+
+    ; Init Input
+    call(Input_Init, player.input)
+    call(OAMManager_Init, player.oam_manager_ptr)
+    txa
+    clc
+    adc #player.oam_manager_ptr
+    tay
+
+    jsr Player_OAMRequest
+
+    plx
+    ply
+    pla
+    rts
+
+Player_OAMRequest:
+    pha
+    phy
+    call_ptr(OAMManager_Request, player.oam_manager_ptr) ; Request 1 OAM object
+
+    ; VRAM address 0 is a transparent tile. 1 is a grass tile in the test.
+    A8
+    lda #1
+    sta oam_object.vram, Y
+    A16
+
+    ; Save the pointer for testing later
+    tya
+    sta player.oam_obj_ptr, X
+
+    ply
+    pla
+    rts
+
+Player_VBlank:
+    pha
+
+    call(Input_VBlank, player.input)
+    jsr Player_Input
+    call_ptr(OAMManager_VBlank, player.oam_manager_ptr)
```

Comment:
OAM Manager's `VBlank` routine is called by the engine. You don't need to call it here. It will manage rendering the OAM space.